graylog-storage-opensearch3/src/main/java/org/graylog/storage/opensearch3/SecurityAdapterOS.java

```diff
@@ -16,97 +16,59 @@
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.annotation.Nonnull;
 import jakarta.inject.Inject;
-import org.graylog.shaded.opensearch2.org.apache.http.HttpEntity;
-import org.graylog.shaded.opensearch2.org.apache.http.entity.ContentType;
-import org.graylog.shaded.opensearch2.org.apache.http.entity.StringEntity;
-import org.graylog.shaded.opensearch2.org.opensearch.client.Request;
 import org.graylog2.indexer.security.SecurityAdapter;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.opensearch.security.GetRoleMappingResponse;
+import org.opensearch.client.opensearch.security.PatchRoleMappingResponse;
+import org.opensearch.client.opensearch.security.RoleMapping;
 
-import java.util.ArrayList;
+import java.io.IOException;
 import java.util.List;
 
 /**
  * see: https://opensearch.org/docs/latest/security/access-control/api/
  */
 public class SecurityAdapterOS implements SecurityAdapter {
-    private static final Logger LOG = LoggerFactory.getLogger(SecurityAdapterOS.class);
 
-    final private PlainJsonApi jsonApi;
-    private final ObjectMapper objectMapper;
+    private final OfficialOpensearchClient client;
 
     @Inject
-    public SecurityAdapterOS(final ObjectMapper objectMapper,
-                             final PlainJsonApi jsonApi) {
-        this.objectMapper = objectMapper;
-        this.jsonApi = jsonApi;
+    public SecurityAdapterOS(OfficialOpensearchClient openSearchClient) {
+        this.client = openSearchClient;
     }
 
     @Override
     public Mapping getMappingForRole(final String role) {
-        final JsonNode result = jsonApi.perform(request("GET", "rolesmapping/" + role), "Unable to retrieve role mapping for role " + role);
-
-        final JsonNode fields = result.path(role);
         try {
-            return objectMapper.treeToValue(fields, Mapping.class);
-        } catch (JsonProcessingException e) {
+            final GetRoleMappingResponse response = client.sync().security().getRoleMapping(r -> r.role(role));
+            final RoleMapping roleMapping = response.get(role);
+            return new Mapping(roleMapping.backendRoles(), roleMapping.hosts(), roleMapping.users());
+        } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
     public MappingResponse addUserToRoleMapping(final String role, final String user) {
-        final var mapping = getMappingForRole(role);
-        if(!mapping.users().contains(user)) {
-            final List<String> users = new ArrayList<>();
-            users.addAll(mapping.users());
-            users.add(user);
-            return setUserToRoleMapping(role, new Mapping(mapping.backendRoles(), mapping.hosts(), users));
-        } else {
+        if (getMappingForRole(role).users().contains(user)) {
             return new MappingResponse("OK", "User already in mapping");
```

Comment:
It is a good candidate for a static constant in `MappingResponse` class, I guess.